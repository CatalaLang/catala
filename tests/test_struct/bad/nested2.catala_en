## Article

```catala
declaration structure S:
  data x content E
  data y content integer

declaration enumeration E:
  -- Case1 content boolean
  -- Case2 content S

declaration scope A:
  context x content E
```
