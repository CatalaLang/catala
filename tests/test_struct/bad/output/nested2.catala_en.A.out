[ERROR] Cyclic dependency detected between types!
[ERROR] 
[ERROR] Cycle type S, declared:
[ERROR]  --> test_struct/bad/nested2.catala_en
[ERROR]   | 
[ERROR] 4 | declaration structure S:
[ERROR]   |                       ^
[ERROR]   + Article
[ERROR] 
[ERROR] Used here in the definition of another cycle type E:
[ERROR]   --> test_struct/bad/nested2.catala_en
[ERROR]    | 
[ERROR] 10 |   -- Case2 content S
[ERROR]    |                    ^
[ERROR]    + Article
[ERROR] 
[ERROR] Cycle type E, declared:
[ERROR]  --> test_struct/bad/nested2.catala_en
[ERROR]   | 
[ERROR] 8 | declaration enumeration E:
[ERROR]   |                         ^
[ERROR]   + Article
[ERROR] 
[ERROR] Used here in the definition of another cycle type S:
[ERROR]  --> test_struct/bad/nested2.catala_en
[ERROR]   | 
[ERROR] 5 |   data x content E
[ERROR]   |                  ^
[ERROR]   + Article
